src/cdk/explorer/cdkProject.ts

```diff
@@ -4,22 +4,21 @@
  */
 
 import * as vscode from 'vscode'
-import { readFileAsString } from '../../shared/filesystemUtilities'
+import { SystemUtilities } from '../../shared/systemUtilities'
 import { ConstructTree } from './tree/types'
 
 export interface CdkApp {
     location: CdkAppLocation
-    metadata: ConstructTree
+    constructTree: ConstructTree
 }
 
 export interface CdkAppLocation {
-    workspaceFolder: vscode.WorkspaceFolder
-    cdkJsonPath: string
-    treePath: string
+    cdkJsonUri: vscode.Uri
```

Comment:
URI should always be preferred for absolute paths, even when we 'know' that the file is on disk.